File: 2025/SoyoungJung/week4/2025-06-29-Guess-Number-Higher-or-Lower.ts
const target = 42;

function guess(num: number): -1 | 0 | 1 {
  if (num === target) return 0;
  if (num > target) return -1;
  return 1;
}

function guessNumber(n: number): number {
  let low = 1;
  let high = n;

  while (low <= high) {
    const mid = Math.floor(low + (high - low) / 2);
    const result = guess(mid);

    if (result === 0) {
      return mid;
    } else if (result === -1) {
      high = mid - 1;
    } else {
      low = mid + 1;
    }
  }

  return -1;
}

File: 2025-06-29-Guess-Number-Higher-or-Lower.py
# The guess API is already defined for you.
# @param num, your guess
# @return -1 if num is higher than the picked number
#          1 if num is lower than the picked number
#          otherwise return 0
# def guess(num):

class Solution(object):
    def guessNumber(self, n):
        """
        :type n: int
        :rtype: int
        """
        left, right = 0, n

        while True:
            mid = (left + right) // 2
            num = guess(mid)

            if num == -1 :
                right = mid -1
            elif num == 0:
                return mid
            else :
                left = mid + 1

